sdk/go/pulumi/context.go

```diff
@@ -1851,6 +2090,23 @@ func (ctx *Context) RegisterStackTransformation(t ResourceTransformation) error
 	return nil
 }
 
+// XRegisterStackTransform adds a transform to all future resources constructed in this Pulumi stack.
+//
+// Experimental.
+func (ctx *Context) XRegisterStackTransform(t XResourceTransform) error {
+	if !ctx.state.supportsTransforms {
+		return fmt.Errorf("update engine")
```

Comment:
Should we have a nicer error message here? Something like "upgrade your version of the Pulumi CLI". We have other errors elsewhere (e.g. in automation API) that we could use as the basis for what we put here.